deployment/cdk/opensearch-service-migration/lib/service-stacks/reindex-from-snapshot-stack.ts

```diff
@@ -170,40 +172,60 @@ export class ReindexFromSnapshotStack extends MigrationServiceCore {
 
         // Calculate the volume size based on the max shard size
         // Have space for the snapshot and an unpacked copy, with buffer
-        const volumeSizeGB = Math.max(
-            Math.ceil(maxShardSizeBytes/(1000**3) * 2 * 1.15),
+        const shardVolumeSizeGiBBufferMultiple = 1.10
+        const shardVolumeSizeGiB = Math.max(
+            Math.ceil(maxShardSizeGiB * 2 * shardVolumeSizeGiBBufferMultiple),
             1
         )
 
-        if (volumeSizeGB > 16000) {
+        if (shardVolumeSizeGiB > (16*1024)) {
             // 16 TiB is the maximum volume size for GP3
             throw new Error(`"Your max shard size of ${props.maxShardSizeGiB} GiB is too large to migrate."`)
         }
 
-        // Volume we'll use to download and unpack the snapshot
-        const snapshotVolume = new ServiceManagedVolume(this, 'SnapshotVolume', {
-            name: 'snapshot-volume',
-            managedEBSVolume: {
-                size: Size.gibibytes(volumeSizeGB),
-                volumeType: EbsDeviceVolumeType.GP3,
-                fileSystemType: FileSystemType.XFS,
-                throughput: props.reindexFromSnapshotWorkerSize === "maximum" ? 450 : 125,
-                tagSpecifications: [{
-                    tags: {
-                        Name: `rfs-snapshot-volume-${props.stage}`,
-                    },
-                    propagateTags: EbsPropagatedTagSource.SERVICE,
-                }],
-                encrypted: true,
-            },
-        });
+        // Reserve 5 GiB of storage for system
+        const systemStorageGiB = 5
+        let ephemeralStorageGiB = systemStorageGiB
+        if (isStackInGovCloud(this)) {
+            // ECS EBS attachment is not supported in GovCloud
+            // https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/govcloud-ecs.html#govcloud-ecs-diffs
+            // Use Ephemeral Storage instead, adding size for shard
+            ephemeralStorageGiB = Math.ceil(shardVolumeSizeGiB + systemStorageGiB)
```

Comment:
Nit: Lots of math, that logic is easier to test if it was refactored into a stand-alone function(s)